iosApp/Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift

```diff
@@ -62,12 +67,17 @@ struct SurveyDetailView: View {
 
     var surveyQuestionView: some View {
         VStack {
-            if isShowingTitle {
+            if dataSource.isShowingTitle {
                 surveyTitleView
                     .transition(.move(edge: .leading).combined(with: .opacity))
+            } else if let surveyDetail = dataSource.viewState.surveyDetail {
+                if questionIndex.isMultiple(of: 2) {
+                    animatedSurveyQuestionView(surveyDetail: surveyDetail)
+                } else {
+                    animatedSurveyQuestionView(surveyDetail: surveyDetail)
+                }
```

Comment:
Why do we need this check `if questionIndex.isMultiple(of: 2)`? The implementation looks the same to me 🤔

Reply:
This is for the slide left animation where the existing one will slide out to disappear and a new one will slide in.

Reply:
Updated POW with the animation

Reply:
@blyscuit but then the `if` check looks redundant because regardless of the `questionIndex`, it is doing the same `animatedSurveyQuestionView(surveyDetail: surveyDetail)`. Can you check again about the animation logic here? 🤔

Reply:
@minhnimble  Updated with `id`.